lib/src/evaluation_context.dart

```diff
@@ -58,6 +69,17 @@ void warnForDeprecation(String message, Deprecation deprecation) {
   EvaluationContext.current.warn(message, deprecation);
 }
 
+/// Prints a deprecation warning with [message] of type [deprecation],
+/// using stderr if there is no [EvaluationContext.current].
+void warnForDeprecationFromApi(String message, Deprecation deprecation) {
+  var context = EvaluationContext.currentOrNull;
+  if (context == null) {
+    Logger.stderr().warnForDeprecation(deprecation, message);
+  } else {
+    context.warn(message, deprecation);
+  }
```

Comment:
You can do this a bit more cleanly with pattern matching:

```suggestion
  if (EvaluationContext.currentOrNull case var context?) {
    context.warn(message, deprecation);
  } else {
    Logger.stderr().warnForDeprecation(deprecation, message);
  }
```

Reply:
Addressed in https://github.com/sass/dart-sass/pull/2117/commits/63c17e31c3f5f6b168bba3f32ecc33ac65113593